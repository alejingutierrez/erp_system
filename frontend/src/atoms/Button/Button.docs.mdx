import { Meta, Canvas, Story, ArgsTable } from '@storybook/blocks';
import { Button } from './Button';

<Meta title="Atoms/Button" of={Button} />

# Button

The `Button` component triggers an action or event. Use the `variant`, `intent` and `size` props to adjust its style and importance.

<Canvas>
  <Story name="Examples">
    <div className="flex items-center gap-4">
      <Button>Primary</Button>
      <Button intent="secondary">Secondary</Button>
      <Button variant="outline" intent="tertiary">Outline</Button>
    </div>
  </Story>
</Canvas>

<ArgsTable of={Button} />
