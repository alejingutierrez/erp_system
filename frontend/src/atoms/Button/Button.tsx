import * as React from 'react';
import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';

import { Spinner } from '@/atoms/Spinner/Spinner';
import { cn } from '@/lib/utils';

const buttonVariants = cva(
  'group inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-all duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:ring-secondary disabled:pointer-events-none disabled:opacity-50 hover:-translate-y-px hover:shadow-md active:translate-y-0 active:shadow-sm active:scale-[0.98]',
  {
    variants: {
      variant: {
        default: 'border-transparent',
        outline: 'border bg-transparent',
        ghost: 'bg-transparent border-transparent',
        glass: 'bg-surface-glass/80 backdrop-blur-lg border border-white/20 shadow-glass text-foreground',
        icon: 'p-0 border-transparent',
      },
      intent: {
        primary: 'text-primary-foreground',
        secondary: 'text-secondary-foreground',
        tertiary: 'text-tertiary-foreground',
        quaternary: 'text-quaternary-foreground',
        success: 'text-success-foreground',
      },
      size: {
        sm: 'h-9',
        md: 'h-10',
        lg: 'h-11',
      },
    },
    compoundVariants: [
      // Default variant styles
      { variant: 'default', intent: 'primary', className: 'bg-gradient-to-br from-primary to-primary/90' },
      { variant: 'default', intent: 'secondary', className: 'bg-gradient-to-br from-secondary to-secondary/80' },
      { variant: 'default', intent: 'tertiary', className: 'bg-tertiary hover:bg-tertiary/90' },
      { variant: 'default', intent: 'quaternary', className: 'bg-quaternary hover:bg-quaternary/90' },
      { variant: 'default', intent: 'success', className: 'bg-success hover:bg-success/90' },
      { variant: 'default', size: 'sm', className: 'px-4 min-w-[8rem]' },
      { variant: 'default', size: 'md', className: 'px-6 min-w-[10rem]' },
      { variant: 'default', size: 'lg', className: 'px-8 min-w-[12rem]' },
      // Icon variant styles
      { variant: 'icon', size: 'sm', className: 'w-9' },
      { variant: 'icon', size: 'md', className: 'w-10' },
      { variant: 'icon', size: 'lg', className: 'w-11' },
      { variant: 'icon', intent: 'primary', className: 'text-primary' },
      { variant: 'icon', intent: 'secondary', className: 'text-secondary' },
      { variant: 'icon', intent: 'tertiary', className: 'text-tertiary' },
      { variant: 'icon', intent: 'quaternary', className: 'text-quaternary' },
      { variant: 'icon', intent: 'success', className: 'text-success' },
      // Outline variant styles
      { variant: 'outline', intent: 'primary', className: 'border-primary text-primary hover:bg-primary hover:text-primary-foreground' },
      { variant: 'outline', intent: 'secondary', className: 'border-secondary text-secondary hover:bg-secondary hover:text-secondary-foreground' },
      { variant: 'outline', intent: 'tertiary', className: 'border-tertiary text-tertiary hover:bg-tertiary hover:text-tertiary-foreground' },
      { variant: 'outline', intent: 'quaternary', className: 'border-quaternary text-quaternary hover:bg-quaternary hover:text-quaternary-foreground' },
      { variant: 'outline', intent: 'success', className: 'border-success text-success hover:bg-success hover:text-success-foreground' },
      { variant: 'outline', size: 'sm', className: 'px-4 min-w-[8rem]' },
      { variant: 'outline', size: 'md', className: 'px-6 min-w-[10rem]' },
      { variant: 'outline', size: 'lg', className: 'px-8 min-w-[12rem]' },
      // Ghost variant styles
      { variant: 'ghost', intent: 'primary', className: 'text-primary hover:bg-primary/10' },
      { variant: 'ghost', intent: 'secondary', className: 'text-secondary hover:bg-secondary/10' },
      { variant: 'ghost', intent: 'tertiary', className: 'text-tertiary hover:bg-tertiary/10' },
      { variant: 'ghost', intent: 'quaternary', className: 'text-quaternary hover:bg-quaternary/10' },
      { variant: 'ghost', intent: 'success', className: 'text-success hover:bg-success/10' },
      { variant: 'ghost', size: 'sm', className: 'px-4 min-w-[8rem]' },
      { variant: 'ghost', size: 'md', className: 'px-6 min-w-[10rem]' },
      { variant: 'ghost', size: 'lg', className: 'px-8 min-w-[12rem]' },
      // Glass variant spacing
      { variant: 'glass', size: 'sm', className: 'px-4 min-w-[8rem]' },
      { variant: 'glass', size: 'md', className: 'px-6 min-w-[10rem]' },
      { variant: 'glass', size: 'lg', className: 'px-8 min-w-[12rem]' },
    ],
    defaultVariants: {
      variant: 'default',
      intent: 'primary',
      size: 'md',
    },
  },
);

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {
  asChild?: boolean;
  isLoading?: boolean;
  loadingText?: string;
  LeftIcon?: React.ElementType;
  RightIcon?: React.ElementType;
}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      className,
      intent,
      size,
      variant,
      asChild = false,
      isLoading = false,
      loadingText,
      LeftIcon,
      RightIcon,
      children,
      ...props
    },
    ref,
  ) => {
    const Comp = asChild ? Slot : 'button';
    const iconSize = size === 'lg' ? 20 : size === 'md' ? 18 : 16;
    const Icon = LeftIcon || RightIcon; // For icon-only variant

    const leftIconAnimation = 'transition-transform duration-300 ease-out group-hover:-translate-x-1';
    const rightIconAnimation = 'transition-transform duration-300 ease-out group-hover:translate-x-1';

    return (
      <Comp
        className={cn(buttonVariants({ intent, size, variant, className }))}
        ref={ref}
        disabled={isLoading || props.disabled}
        {...props}
      >
        {isLoading ? (
          <div className="flex items-center justify-center">
            <Spinner size={size} />
            {loadingText && <span className="ml-2 animate-pulse">{loadingText}</span>}
          </div>
        ) : variant === 'icon' ? (
          Icon && <Icon size={iconSize} />
        ) : (
          <>
            {LeftIcon && <LeftIcon className={cn('mr-2', leftIconAnimation)} size={iconSize} />}
            {children}
            {RightIcon && <RightIcon className={cn('ml-2', rightIconAnimation)} size={iconSize} />}
          </>
        )}
      </Comp>
    );
  },
);
Button.displayName = 'Button';

export { Button, buttonVariants };
