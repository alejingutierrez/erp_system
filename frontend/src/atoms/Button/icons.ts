import {
  AlertCircle,
  Check,
  ChevronDown,
  ChevronLeft,
  ChevronRight,
  ChevronUp,
  Copy,
  Download,
  Edit,
  File,
  Folder,
  Home,
  LogIn,
  LogOut,
  Mail,
  Minus,
  MoreHorizontal,
  Plus,
  Save,
  Search,
  Settings,
  Trash2,
  Upload,
  User,
  X,
} from 'lucide-react';

export const iconMap = {
  AlertCircle,
  Check,
  ChevronDown,
  ChevronLeft,
  ChevronRight,
  ChevronUp,
  Copy,
  Download,
  Edit,
  File,
  Folder,
  Home,
  LogIn,
  LogOut,
  Mail,
  Minus,
  MoreHorizontal,
  Plus,
  Save,
  Search,
  Settings,
  Trash2,
  Upload,
  User,
  X,
};

export type IconName = keyof typeof iconMap;
