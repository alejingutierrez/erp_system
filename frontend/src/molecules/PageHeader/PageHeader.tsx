import * as React from 'react';
import { cn } from '@/lib/utils';
import { Breadcrumbs, type BreadcrumbItem } from '@/atoms/Breadcrumbs';
import { Heading } from '@/atoms/Heading';
import { Icon, type IconName } from '@/atoms/Icon';
import { Divider } from '@/atoms/Divider';

export interface PageHeaderProps extends React.HTMLAttributes<HTMLElement> {
  /** Main title content */
  title: React.ReactNode;
  /** Breadcrumb navigation items */
  breadcrumbs?: BreadcrumbItem[];
  /** Secondary action elements */
  actions?: React.ReactNode;
  /** Optional icon to show next to title */
  icon?: IconName;
  /** Render divider below header */
  divider?: boolean;
}

export const PageHeader = React.forwardRef<HTMLElement, PageHeaderProps>(
  (
    { title, breadcrumbs, actions, icon, divider = false, className, ...props },
    ref,
  ) => {
    return (
      <header
        ref={ref}
        className={cn('page-header space-y-2', className)}
        {...props}
      >
        {breadcrumbs && breadcrumbs.length > 0 && (
          <Breadcrumbs items={breadcrumbs} className="mb-1" />
        )}
        <div className="flex flex-col gap-2 sm:flex-row sm:items-center sm:justify-between">
          <div className="title-group flex items-center gap-2">
            {icon && <Icon name={icon} className="h-6 w-6 text-muted-foreground" aria-hidden="true" />}
            <Heading level={2}>{title}</Heading>
          </div>
          {actions && <div className="actions flex flex-wrap items-center gap-2">{actions}</div>}
        </div>
        {divider && <Divider className="mt-2" />}
      </header>
    );
  },
);
PageHeader.displayName = 'PageHeader';

export default PageHeader;
