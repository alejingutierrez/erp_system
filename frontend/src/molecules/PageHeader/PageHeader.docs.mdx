import { Meta, Canvas, Story, ArgsTable } from '@storybook/blocks';
import { PageHeader } from './PageHeader';
import { Button } from '@/atoms/Button';

<Meta title="Molecules/PageHeader" of={PageHeader} />

# PageHeader

The `PageHeader` component displays breadcrumbs, a main title and
an area for secondary actions. It adjusts actions below the title on
small screens and can optionally show a divider.

<Canvas>
  <Story name="Structure">
    <PageHeader
      title="Products"
      breadcrumbs={[{ label: 'Home', href: '#' }, { label: 'Catalog', href: '#' }, { label: 'Products' }]}
      actions={<Button intent="secondary">New</Button>}
      icon="Folder"
      divider
    />
  </Story>
</Canvas>

<Canvas>
  <Story name="Many actions">
    <PageHeader
      title="Orders"
      breadcrumbs={[{ label: 'Home', href: '#' }, { label: 'Orders' }]}
      actions={
        <div className="flex gap-2 flex-wrap">
          <Button intent="secondary">Export</Button>
          <Button intent="secondary">Filter</Button>
          <Button intent="secondary">New</Button>
        </div>
      }
    />
  </Story>
</Canvas>

<ArgsTable of={PageHeader} />
