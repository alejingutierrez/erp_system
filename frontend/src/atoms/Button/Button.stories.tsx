import type { Meta, StoryObj } from '@storybook/react';

import { Button, ButtonProps } from './Button';
import { iconMap, IconName } from './icons';

// Define a custom type for the story's args that includes our control-only props
interface ButtonStoryProps extends ButtonProps {
  leftIconName?: IconName;
  rightIconName?: IconName;
}

const iconOptions = Object.keys(iconMap) as IconName[];

const meta: Meta<ButtonStoryProps> = {
  title: 'Atoms/Button',
  component: Button,
  tags: ['autodocs'],
  argTypes: {
    variant: { control: 'select', options: ['default', 'outline', 'ghost', 'glass', 'icon'] },
    intent: { control: 'select', options: ['primary', 'secondary', 'tertiary', 'quaternary', 'success'] },
    size: { control: 'select', options: ['sm', 'md', 'lg'] },
    isLoading: { control: 'boolean' },
    loadingText: { control: 'text', if: { arg: 'isLoading', eq: true } },
    disabled: { control: 'boolean' },
    children: { control: 'text' },
    onClick: { action: 'clicked', table: { category: 'Events' } },
    leftIconName: { name: 'Left Icon', control: { type: 'select' }, options: [undefined, ...iconOptions], table: { category: 'Icons' } },
    rightIconName: { name: 'Right Icon', control: { type: 'select' }, options: [undefined, ...iconOptions], table: { category: 'Icons' } },
    LeftIcon: { table: { disable: true } },
    RightIcon: { table: { disable: true } },
    asChild: { table: { disable: true } },
  },
  render: ({ leftIconName, rightIconName, ...args }) => {
    const LeftIcon = leftIconName ? iconMap[leftIconName] : undefined;
    const RightIcon = rightIconName ? iconMap[rightIconName] : undefined;
    return <Button {...args} LeftIcon={LeftIcon} RightIcon={RightIcon} />;
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

// == Intents ==
export const Primary: Story = { args: { intent: 'primary', children: 'Primary Button' } };
export const Secondary: Story = { args: { intent: 'secondary', children: 'Secondary Button' } };
export const Tertiary: Story = { args: { intent: 'tertiary', children: 'Tertiary Button' } };
export const Quaternary: Story = { args: { intent: 'quaternary', children: 'Quaternary Button' } };
export const Success: Story = { args: { intent: 'success', children: 'Success Button' } };

// == Variants ==
export const Outline: Story = { args: { variant: 'outline', intent: 'secondary', children: 'Outline' } };
export const Ghost: Story = { args: { variant: 'ghost', intent: 'primary', children: 'Ghost' } };
export const Glass: Story = {
  args: { variant: 'glass', children: 'Glass Button', leftIconName: 'Star' },
  decorators: [
    (Story) => (
      <div
        className="w-full h-48 p-8 flex items-center justify-center rounded-lg"
        style={{
          backgroundImage: 'url(https://images.unsplash.com/photo-1554147090-e1221a04a025?q=80&w=2070&auto=format&fit=crop)',
          backgroundSize: 'cover',
        }}
      >
        <Story />
      </div>
    ),
  ],
};

// == States ==
export const Disabled: Story = { args: { intent: 'primary', disabled: true, children: 'Disabled' } };
export const Loading: Story = { args: { intent: 'primary', isLoading: true, children: 'Hidden' } };
export const LoadingWithText: Story = { args: { intent: 'secondary', isLoading: true, loadingText: 'Saving...', children: 'Hidden' } };

// == Icon Usage ==
export const WithLeftIcon: Story = { args: { intent: 'primary', children: 'Login', leftIconName: 'Mail' } };
export const WithRightIcon: Story = { args: { intent: 'secondary', children: 'Learn More', rightIconName: 'ChevronRight' } };
export const IconOnly: Story = { args: { variant: 'icon', intent: 'primary', leftIconName: 'Heart', 'aria-label': 'Favorite' } };
