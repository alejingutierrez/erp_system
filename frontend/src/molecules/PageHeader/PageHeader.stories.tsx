import type { Meta, StoryObj } from '@storybook/react';
import { PageHeader, type PageHeaderProps } from './PageHeader';
import { Button } from '@/atoms/Button';

const meta: Meta<PageHeaderProps> = {
  title: 'Molecules/PageHeader',
  component: PageHeader,
  tags: ['autodocs'],
  argTypes: {
    title: { control: 'text' },
    breadcrumbs: { control: 'object' },
    actions: { control: 'object' },
    icon: { control: 'text' },
    divider: { control: 'boolean' },
  },
};

export default meta;

type Story = StoryObj<typeof meta>;

const baseCrumbs = [
  { label: 'Home', href: '#' },
  { label: 'Products', href: '#' },
  { label: 'Item' },
];

export const Default: Story = {
  args: {
    title: 'Products',
    breadcrumbs: baseCrumbs,
    divider: true,
  },
};

export const WithActions: Story = {
  args: {
    title: 'Orders',
    breadcrumbs: [{ label: 'Home', href: '#' }, { label: 'Orders' }],
    actions: (
      <div className="flex gap-2 flex-wrap">
        <Button intent="secondary">Export</Button>
        <Button intent="secondary">New</Button>
      </div>
    ),
  },
};

export const MobileCollapse: Story = {
  args: {
    title: 'Mobile view',
    breadcrumbs: [{ label: 'Home', href: '#' }, { label: 'Mobile' }],
    actions: (
      <div className="flex gap-2 flex-wrap">
        <Button intent="secondary">Export</Button>
        <Button intent="secondary">Filter</Button>
        <Button intent="secondary">New</Button>
      </div>
    ),
  },
  parameters: {
    viewport: { defaultViewport: 'mobile1' },
  },
};
