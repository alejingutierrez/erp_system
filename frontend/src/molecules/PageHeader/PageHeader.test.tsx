import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { PageHeader } from './PageHeader';
import { Button } from '@/atoms/Button';

const breadcrumbs = [
  { label: 'Home', href: '#' },
  { label: 'Products' },
];

describe('PageHeader', () => {
  it('renders breadcrumbs and title', () => {
    render(<PageHeader title="Products" breadcrumbs={breadcrumbs} />);
    expect(screen.getByRole('heading', { name: 'Products' })).toBeInTheDocument();
    expect(screen.getByRole('navigation', { name: 'Breadcrumb' })).toBeInTheDocument();
  });

  it('renders actions', () => {
    render(
      <PageHeader
        title="Products"
        breadcrumbs={breadcrumbs}
        actions={<Button>New</Button>}
      />,
    );
    expect(screen.getByRole('button', { name: 'New' })).toBeInTheDocument();
  });

  it('matches mobile snapshot', () => {
    const { asFragment } = render(
      <PageHeader
        title="Mobile"
        breadcrumbs={breadcrumbs}
        actions={<Button>New</Button>}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
