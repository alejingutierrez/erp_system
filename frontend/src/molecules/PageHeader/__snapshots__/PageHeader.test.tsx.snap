// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeader > matches mobile snapshot 1`] = `
<DocumentFragment>
  <header
    class="page-header space-y-2"
  >
    <nav
      aria-label="Breadcrumb"
      class="text-sm mb-1"
    >
      <ol
        class="flex flex-wrap items-center"
      >
        <li
          class="flex items-center"
        >
          <a
            class="text-sm font-medium underline underline-offset-4 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-background text-secondary hover:text-secondary/80 whitespace-nowrap"
            href="#"
            style="text-decoration: underline;"
          >
            Home
          </a>
          <span
            class="mx-2 text-muted-foreground"
          >
            /
          </span>
        </li>
        <li
          class="flex items-center"
        >
          <span
            aria-current="page"
            class="font-medium text-foreground whitespace-nowrap"
          >
            Products
          </span>
        </li>
      </ol>
    </nav>
    <div
      class="flex flex-col gap-2 sm:flex-row sm:items-center sm:justify-between"
    >
      <div
        class="title-group flex items-center gap-2"
      >
        <h2
          class="font-heading text-4xl font-medium tracking-tight text-left text-primary"
        >
          Mobile
        </h2>
      </div>
      <div
        class="actions flex flex-wrap items-center gap-2"
      >
        <button
          class="group inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-all duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:ring-secondary disabled:pointer-events-none disabled:opacity-50 hover:-translate-y-px hover:shadow-sm active:translate-y-0 active:shadow-md active:scale-105 border-transparent text-primary-foreground h-10 bg-gradient-to-br from-primary to-primary/90 hover:from-primary/90 hover:to-primary/80 px-6 min-w-[10rem]"
        >
          New
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
